src/client/src/Stream.c

```diff
@@ -2079,38 +2096,28 @@ STATUS resetCurrentViewItemStreamStart(PKinesisVideoStream pKinesisVideoStream)
     pKinesisVideoClient->clientCallbacks.lockMutexFn(pKinesisVideoClient->clientCallbacks.customData, pKinesisVideoClient->base.lock);
     clientLocked = TRUE;
 
-    // Get the required size for the cluster header
-    // NOTE: Stream start is only on a cluster boundary
-    CHK_STATUS(mkvgenGetMkvOverheadSize(pKinesisVideoStream->pMkvGenerator,
-                                        MKV_STATE_START_CLUSTER,
-                                        &clusterHeaderSize));
+    // As we are removing the MKV header, the resulting allocation size will be actually smaller
+    // We will simply copy/shift the data, including the MKV tags if any and the cluster header
+    // forward and will set the size of the allocation
 
     // Get the existing frame allocation
     CHK_STATUS(heapMap(pKinesisVideoClient->pHeap, pViewItem->handle, (PVOID*) &pFrame, &allocSize));
     CHK(allocSize < MAX_UINT32, STATUS_INVALID_ALLOCATION_SIZE);
-    packagedSize = (UINT32) allocSize;
+
+    packagedSize = pViewItem->length;
```

Comment:
I think we should have a check that makes sure that `pViewItem->length <= allocSize` since this might cause an unexpected crash.

Reply:
I think so too indeed. The same should be with the previous case in getStreamData